### PR TITLE
[UIE-194] First pass at Modal storybook

### PR DESCRIPTION
**Pull request by cahrens**
### Jira Ticket: https://broadworkbench.atlassian.net/browse/UIE-194

Can be viewed by selecting the branch and most recent build at https://www.chromatic.com/library?appId=65fc89c9335768720ff8605a&buildNumber=736.

### CODE REVIEW

**Comment by cahrens** [comment]
@mlilynolting I played around with this and pushed up some changes. Let me know what you think!

https://65fc89c9335768720ff8605a-qgvozaneqv.chromatic.com/?path=/docs/packages-components-modal--docs

**Comment by mlilynolting** [comment]
@cahrens thank you, your changes look great. Glad to have an example of a way to demonstrate "you can put custom components here" without going overboard

File: src/stories/packages/components/Modal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    okButton: 'Close It',
+    showButtons: true,
+    showCancel: true,
+    showX: true,
+    styles: undefined,
+    title: 'Confirm',
+    titleChildren: undefined,
+    width: 450,
+    onDismiss: () => {},
+  } satisfies ModalProps,
```
Comment (mlilynolting) [inline src/stories/packages/components/Modal.stories.tsx:103]: `satisfies` is a new one for me, love it

**Review by mlilynolting — APPROVED** [review]

File: src/stories/packages/components/Modal.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      control: 'text',
+      description: 'text of Cancel button',
+    },
+    children: {
+      control: 'text',
+      description: 'modal body (string OR components)',
+    },
+    danger: {
+      control: 'boolean',
+      description: 'is the action dangerous?',
```
Comment (nawatts) [inline src/stories/packages/components/Modal.stories.tsx]: Nit: Would "destructive" more clearly describe the intent here?

```suggestion
      description: 'is the action destructive?',
```

**Review by nawatts — APPROVED** [review]

**Pull request merged**